docs/development/build-instructions-gn.md

```diff
@@ -38,12 +47,56 @@ $ gclient sync --with_branch_heads --with_tags
 ```sh
 $ cd src
 $ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
-$ gn gen out/Default --args='root_extra_deps=["//electron"] is_electron_build=true is_component_build=true use_jumbo_build=true v8_promise_internal_field_count=1 v8_typed_array_max_size_in_heap=0'
+$ gn gen out/Default --args='target_os="mac"'
 ```
 
-This will generate all the ninja files needed for the build. You shouldn't have
-to run `gn gen` again—if you want to change the build arguments you can run `gn
-args out/Default` to bring up an editor.
+This will generate a build directory `out/Default` under `src/` with
+default chromium build configuration. You can replace `Default` with
+another name, but it should be a subdirectory of `out`. Here `target_os`
+indicates the primary operating system used for the build, you can run
+`gn help target_os` to get list of possible values. Also, to know the list
+of available configuration options, run `gn args out/Default --list`.
+To build `electron` we need some additional arguments depending on the build,
+which are specified below. Also you shouldn't have to run `gn gen` again—if
+you want to change the build arguments, you can run `gn args out/Default` to
+bring up an editor.
+
+* #### GN args for Debug/Component build of Electron:
+
+```gn
+root_extra_deps=["//electron"]
+is_electron_build=true
+is_component_build=true
+use_jumbo_build=true
+is_debug=true
+symbol_level=2
+use_allocator="none"
+enable_widevine = true
+is_component_ffmpeg=true
+ffmpeg_branding="Chrome"
+v8_promise_internal_field_count=1
+v8_typed_array_max_size_in_heap=0
+```
+
+* #### GN args for Release/Non-Component build of Electron:
+
+```gn
+root_extra_deps=["//electron"]
+is_electron_build=true
+is_electron_gn_build=true
+is_component_build=false
+use_jumbo_build=true
+is_official_build=true
+enable_widevine = true
+is_component_ffmpeg=true
+ffmpeg_branding="Chrome"
+v8_promise_internal_field_count=1
+v8_typed_array_max_size_in_heap=0
+
+if (target_os == "win") {
```

Comment:
if you don't set `target_os` in `args.gn`, it will be `""`. `BUILDCONFIG.gn` detects the empty string and changes it to `host_os`. So this `if` won't work how you expect. IMO the `use_allocator_shim` logic should be moved to a libcc patch.